[docs] Add a recipe for drag and drop row grouping (#17638)
mui/mui-x: docs/data/data-grid/components/toolbar/toolbar.md

{{"demo": "GridToolbarFilterBar.js", "bg": "inline", "defaultCodeOpen": false}}

> ### Row grouping bar

Reviewer: More understandable and more likely to be searched?
```suggestion
### Row grouping toolbar
```

The demo below shows how to add a custom toolbar that enables creating and reordering groups with drag and drop.

> Users can drag column headers into the toolbar to create groups, then reorder them by dragging the chips within the toolbar. Groups can be removed by clicking the delete button.

Reviewer: ```suggestion
Users can create groups by dragging column headers into the toolbar, reorder them by dragging the chips within the toolbar, and remove them by clicking the delete button.
```
